### PR TITLE
Feature/p2p sync with push msgs

### PR DESCRIPTION
**Pull request by LukasGasior1**

### CODE REVIEW

**Review by siy — APPROVED** [review]
aLGTM

File: radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/NodeNetworkMessagesModule.java [hunk 1]
```diff
@@ -102,6 +99,11 @@ private SimulatedNetworkImpl network(@Self BFTNode node) {
 		return network.remoteEventDispatcher(StatusResponse.class);
 	}
 
+	@Provides
+	private RemoteEventDispatcher<LedgerStatusUpdate> ledgerStatusUpdateDispatcher(SimulatedNetworkImpl network) {
+		return network.remoteEventDispatcher(LedgerStatusUpdate.class);
+	}
+
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/NodeNetworkMessagesModule.java:106]: Let's start using more generic way of creating these dispatchers e.g.:
```
@ProvidesIntoSet
private RxRemoteDispatcher<?> voteDispatcher(SimulatedNetworkImpl network) {
	return RxRemoteDispatcher.create(Vote.class, network.remoteEventDispatcher(Vote.class));
}
```

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/RemoteSyncService.java [hunk 2]
```diff
@@ -133,6 +155,31 @@ private void processLedgerUpdate(LedgerUpdate ledgerUpdate) {
 		final VerifiedLedgerHeaderAndProof updatedHeader = ledgerUpdate.getTail();
 		if (accComparator.compare(updatedHeader.getAccumulatorState(), this.currentHeader.getAccumulatorState()) > 0) {
 			this.currentHeader = updatedHeader;
+			ledgerUpdate.getNextValidatorSet().ifPresent(newValidatorSet -> this.currentValidatorSet = newValidatorSet);
+			this.sendStatusUpdateToSomePeers(updatedHeader);
+		}
+	}
+
+	private void sendStatusUpdateToSomePeers(VerifiedLedgerHeaderAndProof header) {
+		if (!(this.localSyncService.getSyncState() instanceof SyncState.IdleState)) {
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/RemoteSyncService.java:164]: Do we need this check?

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/RemoteSyncService.java [hunk 3]
```diff
@@ -133,6 +155,31 @@ private void processLedgerUpdate(LedgerUpdate ledgerUpdate) {
...
+			return; // not sending any updates if the node is syncing itself
 		}
+
+		final var statusUpdate = LedgerStatusUpdate.create(header);
+
+		final var nonValidatorPeers = this.peersView.peers().stream()
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/RemoteSyncService.java:170]: Not sure if we need to only send to non-validating peers. It is helpful to have the sync layer just be an overall sync layer.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/messages/remote/LedgerStatusUpdate.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+package com.radixdlt.sync.messages.remote;
+
+import com.radixdlt.consensus.VerifiedLedgerHeaderAndProof;
+
+import java.util.Objects;
+
+/**
+ * A message pushed to a subset of connected non-validator nodes indicating that the ledger state has been updated.
+ */
+public final class LedgerStatusUpdate {
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/messages/remote/LedgerStatusUpdate.java:27]: Spaces vs. Tabs.

**Review by talekhinezh — CHANGES_REQUESTED** [review]

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/RemoteSyncService.java (hunk 2 above)
Comment (LukasGasior1) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/RemoteSyncService.java:164]: this is not strictly needed, but I think it doesn't make much sense to broadcast the status message if the node is still syncing itself

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/RemoteSyncService.java (hunk 3 above)
Comment (LukasGasior1) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/RemoteSyncService.java:170]: validators don't need that, they can just sync up based on received votes/proposals

**Pull request merged**